src/Sylius/Component/Core/Taxation/Applicator/CompositeOrderTaxesCalculationStrategy.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Taxation\Applicator;
+
+use Sylius\Bundle\SettingsBundle\Model\Settings;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+
+/**
+ * @author Mark McKelvie <mark.mckelvie@reiss.com>
+ */
+class CompositeOrderTaxesCalculationStrategy implements OrderTaxesApplicatorInterface
+{
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var Settings
+     */
+    protected $settings;
+
+    /**
+     * @var OrderTaxesApplicatorInterface[]
+     */
+    protected $applicators;
+
+    /**
+     * @param string $type
+     * @param Settings $settings
+     * @param OrderTaxesApplicatorInterface[] $applicators
+     */
+    public function __construct($type, Settings $settings, array $applicators)
```

Comment:
does `$type` need to be injected? Isn't it dictated by the class? should this be a constant?